packages/raf/src/index.ts

```diff
@@ -1,53 +1,61 @@
-import { createSignal, onMount, onCleanup, createMemo } from "solid-js";
+import { createSignal, onCleanup, createMemo, createEffect } from "solid-js";
 
 export type FPS = number | Function;
-const getFps = (fps: FPS) => (typeof fps === "function" ? fps() : fps);
-const calcFpsInterval = (fps: number) => 1000 / fps;
+const getFps = (targetFps: FPS) => (typeof targetFps === "function" ? targetFps() : targetFps);
+const calcFpsInterval = (targetFps: number) => 1000 / targetFps;
 
 /**
  * Creates a method to support RAF.
  * Based on useRafFN (https://github.com/microcipcip/vue-use-kit/blob/master/src/functions/useRafFn/useRafFn.ts).
  *
  * @param callback The callback to run on a frame
- * @param fps Target frame rate, defaults to 60
- * @param runImmediately Determines if the function could run immediately
- * @returns Returns a signal if currently running as well as star tand stop methods
+ * @param targetFps Target frame rate, defaults to Infinity
+ * @returns Returns a signal if currently running as well as start and stop methods
  *
  * @example
  */
 const createRAF = (
-  callback: (timeElapse: number) => void,
-  fps: FPS = 60,
-  runImmediately = true
+  callback: (timeStamp: number) => void,
+  targetFps: FPS = Infinity
 ): [running: () => boolean, start: () => void, stop: () => void] => {
+
   const [running, setRunning] = createSignal(false);
-  const fpsInterval = createMemo(() => calcFpsInterval(getFps(fps)));
-  let wasIdle = false;
-  let startTime = 0;
-  let timeElapsed = 0;
-  let timePaused = 0;
-  let timeDelta = 0;
+  const fpsInterval = createMemo(() => calcFpsInterval(getFps(targetFps)));
+
+  let elapsed = 0;
+  let lastRun = 0;
+  let missedBy = 0;
+
+  let interval = fpsInterval();
+  let fps = getFps(targetFps)
+  createEffect(() => {
```

Comment:
```suggestion
  createComputed(() => {
```
If this effect (and the one below) are for setting variables, shouldn't this be a pure computation? Just so the changes happen as soon as possible.
Probably doesn't matter much. just nitpicking while reeding through the code

Reply:
Thanks for your comments, appreciate the suggestion, very welcome and I would love to hear about the differences between the two, because I don't know this is just confusing. When would you use one over the other? Sorry do not rush, thanks

Reply:
From my understanding, right now there are two differences: an effect is batching and runs after mount, while computed is not. There is probably more going on internally, but from the usage standpoint that's it.
You should utilise effects as much as possible for most cases, because batching helps to reduce unnecessary side-effects (like rendering). And will not run on the server for the same reason.
Computed is used for creating own primitives, for setting variables, where you don't have side effects, only outputs as signals.
In this example you're using createEffect – an impure computation – to have a derived value of sorts, so something you'd use createMemo normally – a pure computation – but if you can't, you can use createComputed – also a pure computation.

Reply:
I'd vote to just use `createEffect` if it works, as that will be more inline with potential ways the API could change later. Ryan did mention in Discord that he thought `createComputed` might be a mistake. Having both introduces room for timing differences, that can lead to unexpected results.

I described in https://github.com/solidjs/solid/issues/879 what I think would be the ideal behavior. In that ideal, 

- deferred/batched effects would be a default (not to confuse the traditional JavaScript notion of batching with Solid's concept of `batch`, which is something different),
- `createComputed` and `batch` would both be removed because the new effect behavior would eliminate the need for them,
- all signal reads would _always_ return their current value and never ever a value in the past (confusing),
- and memos would be deferred/batched too (due to being built on effects) except they'd also be lazy in case a current value is needed before a future effect runs.

In this way, we'd eliminate unnecessary excessive effect runs in a consistent way, and we'd also eliminate glitches.

---

The [Qt C++ framework](https://qt.io) (well known, used in many commercial settings, Linux desktops, and cross-platform apps), made the change I described to their reactive system: in a breaking change, all reactions to reactive states were moved to deferred/batched to improve performance and simplify code execution. I am looking for the article they wrote on it.

Reply:
So the only difference between computed/memo and effect will be that the later don't run on the server?
In this case it doesn't really matter, but in most cases, computed/memo should still be the winner for derived values.